setup.py

```diff
@@ -1,38 +1,96 @@
+from __future__ import print_function
+
 import os
+import sys
+
+from Cython.Build import cythonize
+from setuptools import Command, Extension, setup
+
+
+COMPILER_DIRECTIVES = {}
+DEFINE_MACROS = None
+FLAG_COVERAGE = '--cython-coverage'  # custom flag enabling Cython line tracing
+BASEDIR = os.path.abspath(os.path.dirname(__file__))
+NAME = 'cftime'
+CFTIME_DIR = os.path.join(BASEDIR, NAME)
+CYTHON_FNAME = os.path.join(CFTIME_DIR, '_{}.pyx'.format(NAME))
+
+
+class CleanCython(Command):
+    description = 'Purge artifacts built by Cython'
+    user_options = []
 
-from setuptools import Extension, setup
+    def initialize_options(self):
+        pass
 
-rootpath = os.path.abspath(os.path.dirname(__file__))
+    def finalize_options(self):
+        pass
 
+    def run(self):
+        for rpath, _, fnames in os.walk(CFTIME_DIR):
+            for fname in fnames:
+                _, ext = os.path.splitext(fname)
+                if ext in ('.pyc', '.pyo', '.c', '.so'):
+                    artifact = os.path.join(rpath, fname)
+                    if os.path.exists(artifact):
+                        print('clean: removing file {!r}'.format(artifact))
+                        os.remove(artifact)
+                    else:
+                        print('clean: skipping file {!r}'.format(artifact))
 
-def extract_version(module='cftime'):
+
+def extract_version():
     version = None
-    fname = os.path.join(rootpath, module, '_cftime.pyx')
-    with open(fname) as f:
-        for line in f:
+    with open(CYTHON_FNAME) as fi:
+        for line in fi:
             if (line.startswith('__version__')):
                 _, version = line.split('=')
                 version = version.strip()[1:-1]  # Remove quotation characters.
                 break
     return version
 
 
-with open('requirements.txt') as f:
-    reqs = f.readlines()
-install_requires = [req.strip() for req in reqs]
+def load(fname):
+    result = []
+    with open(fname, 'r') as fi:
+        result = [package.strip() for package in fi.readlines()]
+    return result
+
+
+def description():
+    fname = os.path.join(BASEDIR, 'README.md')
+    with open(fname, 'r') as fi:
+        result = ''.join(fi.readlines())
+    return result
+
+
+if FLAG_COVERAGE in sys.argv:
+    COMPILER_DIRECTIVES = {'linetrace': True,
+                           'warn.maybe_uninitialized': False,
+                           'warn.unreachable': False,
+                           'warn.unused': False}
```

Comment:
Turning on the Cython `linetrace` directive makes the building phase pretty noisy... this is my attempt to calm it down a wee bit by ignoring the verbose Cython based compiler warnings.